galley/cmd/galley/cmd/server.go

```diff
@@ -96,104 +96,104 @@ func serverCmd() *cobra.Command {
+	svr.PersistentFlags().StringVar(&serverArgs.ValidationArgs.DeploymentAndServiceNamespace, "deployment-namespace", "istio-system",
 		"Namespace of the deployment for the validation pod")
-	serverCmd.PersistentFlags().StringVar(&serverArgs.ValidationArgs.DeploymentName, "deployment-name", "istio-galley",
+	svr.PersistentFlags().StringVar(&serverArgs.ValidationArgs.DeploymentName, "deployment-name", "istio-galley",
 		"Name of the deployment for the validation pod")
-	serverCmd.PersistentFlags().StringVar(&serverArgs.ValidationArgs.ServiceName, "service-name", "istio-galley",
+	svr.PersistentFlags().StringVar(&serverArgs.ValidationArgs.ServiceName, "service-name", "istio-galley",
 		"Name of the validation service running in the same namespace as the deployment")
-	serverCmd.PersistentFlags().StringVar(&serverArgs.ValidationArgs.WebhookName, "webhook-name", "istio-galley",
+	svr.PersistentFlags().StringVar(&serverArgs.ValidationArgs.WebhookName, "webhook-name", "istio-galley",
 		"Name of the k8s validatingwebhookconfiguration")
 
 	// Hidden, file only flags for validation specific TLS
-	serverCmd.PersistentFlags().StringVar(&serverArgs.ValidationArgs.CertFile, "validation.tls.clientCertificate", "",
+	svr.PersistentFlags().StringVar(&serverArgs.ValidationArgs.CertFile, "validation.tls.clientCertificate", "",
 		"File containing the x509 Certificate for HTTPS validation.")
-	_ = serverCmd.PersistentFlags().MarkHidden("validation.tls.clientCertificate")
-	serverCmd.PersistentFlags().StringVar(&serverArgs.ValidationArgs.KeyFile, "validation.tls.privateKey", "",
+	_ = svr.PersistentFlags().MarkHidden("validation.tls.clientCertificate")
+	svr.PersistentFlags().StringVar(&serverArgs.ValidationArgs.KeyFile, "validation.tls.privateKey", "",
 		"File containing the x509 private key matching --validation.tls.clientCertificate.")
-	_ = serverCmd.PersistentFlags().MarkHidden("validation.tls.privateKey")
-	serverCmd.PersistentFlags().StringVar(&serverArgs.ValidationArgs.CACertFile, "validation.tls.caCertificates", "",
+	_ = svr.PersistentFlags().MarkHidden("validation.tls.privateKey")
+	svr.PersistentFlags().StringVar(&serverArgs.ValidationArgs.CACertFile, "validation.tls.caCertificates", "",
 		"File containing the caBundle that signed the cert/key specified by --validation.tls.clientCertificate and --validation.tls.privateKey.")
-	_ = serverCmd.PersistentFlags().MarkHidden("validation.tls.caCertificates")
+	_ = svr.PersistentFlags().MarkHidden("validation.tls.caCertificates")
 
-	serverArgs.IntrospectionOptions.AttachCobraFlags(serverCmd)
-	loggingOptions.AttachCobraFlags(serverCmd)
-	_ = viper.BindPFlags(serverCmd.PersistentFlags())
+	serverArgs.IntrospectionOptions.AttachCobraFlags(svr)
+	loggingOptions.AttachCobraFlags(svr)
+	_ = viper.BindPFlags(svr.PersistentFlags())
 
 	cobra.OnInitialize(setupAliases)
 
-	return serverCmd
+	return svr
 }
 
 func setupAliases() {
```

Comment:
While you're here, can you please make sure that the viper aliases are correctly all there? At a glance it looks like some of them are missing, e.g. enableAnalysis.

Reply:
Analysis is there:
https://github.com/istio/istio/blob/ff382fe4ceed2bd5bdd57ed1f7b5adf69c6a7566/galley/cmd/galley/cmd/server.go#L213

Reply:
Whoops, I picked the wrong example.

I'm not seeing useOldProcessor or enableServiceDiscovery, though.